### PR TITLE
Typeorm

### PR DESCRIPTION
**Pull request by spaenleh**

### CODE REVIEW

**Comment by spaenleh** [comment]
This PR includes changes from #215 and super-seeds it as it is based on the ts-vite move, where #215 is based on the old architecture (pre-vite).

File: .eslintrc [hunk 1]
```diff
@@ -66,42 +82,91 @@
...
+          }
+        ],
+        // "@typescript-eslint/explicit-function-return-type": [
+        //   "error",
+        //   {
+        //     "allowExpressions": true,
+        //     "allowTypedFunctionExpressions": true,
+        //     "allowHigherOrderFunctions": true,
+        //   }
+        // ]
```
Comment (pyphilia) [inline .eslintrc:129]: Remove?

File: cypress/e2e/header.cy.ts [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
-
 import { HOME_PATH } from '../../src/config/paths';
 import {
   HEADER_MEMBER_MENU_BUTTON_ID,
-  HEADER_MEMBER_MENU_SEE_PROFILE_BUTTON_ID,
-  HEADER_MEMBER_MENU_SIGN_IN_BUTTON_ID,
-  HEADER_MEMBER_MENU_SIGN_OUT_BUTTON_ID,
-  buildMemberMenuItemId,
+  HEADER_MEMBER_MENU_SEE_PROFILE_BUTTON_ID, // HEADER_MEMBER_MENU_SIGN_IN_BUTTON_ID,
+  HEADER_MEMBER_MENU_SIGN_OUT_BUTTON_ID, // buildMemberMenuItemId,
```
Comment (pyphilia) [inline cypress/e2e/header.cy.ts:5]: Remove comments.

File: cypress/e2e/navigation.cy.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    { memberId: MEMBERS.ANNA.id, permission: PermissionLevel.Read },
+  ],
+});
+describe('Navigation', () => {
+  it('Show navigation on Home', () => {
+    cy.setUpApi({
+      items: [...items, ...sharedItems],
+    });
+    // eslint-disable-next-line no-console
+    console.log(items);
```
Comment (pyphilia) [inline cypress/e2e/navigation.cy.ts:31]: Remove.

File: cypress/support/commands.ts [hunk 4]
```diff
@@ -13,68 +12,69 @@ import {
...
+    mockGetSharedItems({ items, currentMember });
     mockGetItem(
-      { items: cachedItems, currentMember },
+      { items, currentMember },
       getItemError || getCurrentMemberError,
     );
+    mockGetItemChat({ chatMessages });
     mockGetItemMembershipsForItem(items, currentMember);
-    mockGetPublicItem({ items: cachedItems });
+    // mockGetPublicItem({ items: cachedItems });
```
Comment (pyphilia) [inline cypress/support/commands.ts:56]: Remove comments.

File: cypress/support/server.ts [hunk 5]
```diff
@@ -27,52 +40,91 @@ export const MEMBER_PROFILE_PATH = `${Cypress.env(
...
-  }
-  return undefined;
-};
+// const checkIsPublic = (item: MockItem): { statusCode: number } | undefined => {
+//   const isPublic = item?.tags?.find(({ tagId }) => tagId === PUBLIC_TAG_ID);
+//   if (!isPublic) {
+//     return { statusCode: StatusCodes.UNAUTHORIZED };
+//   }
+//   return undefined;
+// };
```
Comment (pyphilia) [inline cypress/support/server.ts:127]: Remove.

File: cypress/support/server.ts [hunk 6]
```diff
@@ -201,38 +305,38 @@ export const mockGetItemMembershipsForItem = (
...
-      }
-
-      return reply({
-        body: item,
-        statusCode: StatusCodes.OK,
-      });
-    },
-  ).as('getPublicItem');
-};
+// export const mockGetPublicItem = (
```
Comment (pyphilia) [inline cypress/support/server.ts:308]: Remove.

File: cypress/support/server.ts [hunk 7]
```diff
@@ -251,47 +355,80 @@ export const mockGetChildren = (items: MockItem[], member: Member): void => {
...
+          checkMemberHasAccess({ item: newItem, items, member }) === undefined,
       );
-      return reply(children);
+      return reply(descendants);
     },
-  ).as('getPublicChildren');
+  ).as('getDescendants');
 };
 
+// export const mockGetPublicChildren = (items: MockItem[]): void => {
```
Comment (pyphilia) [inline cypress/support/server.ts:403]: Remove

File: cypress/support/server.ts [hunk 8]
```diff
@@ -334,72 +471,75 @@ export const mockDefaultDownloadFile = (
...
-      // info: we don't test fixture data anymore since the frontend uses url only
-      if (replyUrl) {
-        return reply({ url: item.filepath });
-      }
-
-      return reply({ fixture: item.filepath });
-    },
-  ).as('publicDownloadFile');
-};
+// export const mockPublicDefaultDownloadFile = (
```
Comment (pyphilia) [inline cypress/support/server.ts:501]: Remove.

File: src/config/queryClient.ts [hunk 9]
```diff
@@ -16,7 +16,8 @@ const {
   DOMAIN,
   // todo: improve types
   notifier: notifier as (e: unknown) => void,
-  enableWebsocket: true,
+  // todo: re-enable once websockets work in refactor
+  enableWebsocket: false,
```
Comment (pyphilia) [inline src/config/queryClient.ts:20]: Set back to true.

File: src/modules/common/HiddenWrapper.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  hidden: boolean;
+  itemId: ItemRecord['id'];
+  children: JSX.Element;
+}): JSX.Element => (
+  <StyledBox
+    isHidden={isHidden}
+    id={buildHiddenWrapperId(itemId, isHidden)}
+    title={
+      isHidden
+        ? "This element is hidden, you can see it because you have admin or write access, users with read access won't see it"
```
Comment (pyphilia) [inline src/modules/common/HiddenWrapper.tsx:32]: Translation?

File: src/modules/item/Item.tsx [hunk 11]
```diff
@@ -38,43 +48,315 @@ import {
...
+      // }
+
+      // // // do not display children folders if they are not pinned
+      // // if (!item.settings?.isPinned) {
+      // //   return null;
+      // // }
+
+      // // only display children folders if they are pinned
+      // if (item.settings?.isPinned) {
+      //   return folderButton;
```
Comment (pyphilia) [inline src/modules/item/Item.tsx:286]: What do we do with this?

File: src/modules/pages/ItemPage.tsx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+  const { mutate: signOut } = useSignOut();
+  const { mutate: itemLoginSignIn } = usePostItemLogin();
+  const { rootId } = useParams();
+  const navigate = useNavigate();
+
+  const ForbiddenContent = <ItemForbiddenScreen />;
+
+  if (!rootId) {
+    navigate(HOME_PATH);
+    // TODO: return not found?
```
Comment (pyphilia) [inline src/modules/pages/ItemPage.tsx:38]: todo?

File: src/modules/tree/DynamicTreeView.tsx [hunk 13]
```diff
@@ -1,99 +1,116 @@
...
+  const [showAll, setShowAll] = useState(false);
+
+  // const {
+  //   data: children,
+  //   isLoading: isLoadingChildren,
+  //   isError: isErrorChildren,
+  // } = useDescendants(rootId, {
+  //   enabled: isFolder,
+  //   getUpdates: isFolder,
+  // });
```
Comment (pyphilia) [inline src/modules/tree/DynamicTreeView.tsx:51]: remove.

File: src/modules/tree/DynamicTreeView.tsx [hunk 14]
```diff
@@ -1,99 +1,116 @@
...
   // types based on TreeView types
   const onToggle = (_event: unknown, nodeIds: string[]) =>
     setExpandedItems(nodeIds);
 
   // show only folder items in the navigation tree
-  const itemsFiltered = items.filter((item) =>
-    GRAASP_MENU_ITEMS.includes(item.type),
+  // const itemsFiltered = items?.filter((item) =>
+  //   GRAASP_MENU_ITEMS.includes(item.type),
+  // );
```
Comment (pyphilia) [inline src/modules/tree/DynamicTreeView.tsx:69]: Remove?

File: vite.config.ts [hunk 15]
```diff
@@ -13,7 +13,7 @@ export default ({ mode }: { mode: string }): UserConfigExport => {
     base: '',
     server: {
       port: parseInt(process.env.VITE_PORT, 10),
-      open: true,
+      // open: true,
```
Comment (pyphilia) [inline vite.config.ts:16]: Remove? Set back?

**Review by pyphilia — APPROVED** [review]
Thanks for the work!! 🌻

**Pull request merged**